17thblog/logIn.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+include('config/config.php');
+include('config/init.php');
+
+echoHeader("Log In", "Log In");
+
+if(isset($_SESSION['Username']) && !empty($_SESSION['Username'])) {
+	echo "
+	<div class='textStyle form'>
+		**You are already logged in as $_SESSION[Username]!**
+		<br>
+		Click here to Log Out:
+		<a href='loggedOut.php'>Log Out</a>
+	</div>";
+}
+
+//form
+$Errors = array();
+
+if(isset($_REQUEST['logIn'])){ //
+
+	validateFormField('Username');
+	validateFormField('Password');
+
+	if(sizeof($Errors) == 0){
+
+		if(verifyUser($_REQUEST['Username'], $_REQUEST['Password']) == true) {
+
+			//set the session username = to that req and set the pass = to that
+			$_SESSION['Username'] = $_REQUEST['Username'];
+			$_SESSION['Password'] = $_REQUEST['Password'];
```

Comment:
There shouldn't be any reason to store the password in the session. Passwords are normally something you want to interact with as little possible just so you don't run the risk of leaking it somewhere else.

Along these lines: You should probably use the UserId as the thing you use to identify a user. They'll log in with a username (humans don't like remembering random numbers for their logins) but internally, their UserId is their identifier. That should be the primary key in the db table, it should be the thing you store in the session, and it should be the parameter you pass to a function to get a user row most of the time.

For an example on why this matters: imagine a user changes their username. You'd have the username stored in the session, but it wouldn't be correct anymore, so they'd potentially be logged in as someone else (if someone else claims their old username). It's best to always use autoincrementing primary keys as the "pointer" to rows in your db.